### PR TITLE
Implemented leaving of Sessions.

### PR DESCRIPTION
**Pull request by Ranray**

### CODE REVIEW

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeaveSessionMessage.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+@MessageEvent(id = 5, handler = PlayerLeaveSessionMessageHandler.class)
+public class PlayerLeaveSessionMessage extends Message {
+
+    private static final int PLAYER_LEAVE_SESSION_MESSAGE = 5;
+
+    private final String username;
+    private final TeamColour colour;
+
+    public PlayerLeaveSessionMessage(String username, TeamColour colour) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeaveSessionMessage.java:20]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/PlayerLeaveSessionMessageHandler.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package nl.soccar.gameserver.message.handler;
+
+import io.netty.buffer.ByteBuf;
+import java.util.logging.Level;
```
Comment (TehMessiah) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/PlayerLeaveSessionMessageHandler.java]: MINOR Remove this unused import 'java.util.logging.Level'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java [hunk 3]
```diff
@@ -130,21 +130,30 @@ private void sendJoinedSessionNotification(Player joinedPlayer, TeamColour joine
...
-        PlayerLeftSessionMessage m = new PlayerLeftSessionMessage(player.getUsername(), team.getTeamColour());
+        room.getAllPlayers().stream().map(server::getConnectionFromPlayer).forEach(c -> System.out.println(c));
+        
+        PlayerLeftSessionMessage m = new PlayerLeftSessionMessage(player.getUsername(), colour);
         room.getAllPlayers().stream().map(server::getConnectionFromPlayer).forEach(c -> c.send(m));
+        LOGGER.log(Level.INFO, "Player {0} left the Session.", player.getUsername());
         return true;
     }
+    
+    public void leaveSession(Player player, Session session) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java:147]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java [hunk 4]
```diff
@@ -130,21 +130,30 @@ private void sendJoinedSessionNotification(Player joinedPlayer, TeamColour joine
...
-
-        Team team = teamBlue.getPlayers().stream().filter(p -> p.getUsername().equals(player.getUsername())).count() > 0 ? teamBlue : teamRed;
+        
+        Team team = colour == TeamColour.BLUE ? room.getTeamBlue() : room.getTeamRed();
         team.leave(player);
 
         player.setCurrentSession(null);
 
-        PlayerLeftSessionMessage m = new PlayerLeftSessionMessage(player.getUsername(), team.getTeamColour());
+        room.getAllPlayers().stream().map(server::getConnectionFromPlayer).forEach(c -> System.out.println(c));
```
Comment (TehMessiah) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS106)
MINOR Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1612)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 6 issues
* MAJOR 1 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

**Comment by ghost** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeftSessionMessage.java [hunk 5]
```diff
@@ -30,7 +30,7 @@ public TeamColour getTeam() {
 
     @Override
     public int getId() {
-        return 3;
+        return 4;
```
Comment (BeeHiveJava) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeftSessionMessage.java:33]: Constante toevoegen.

**Review by BeeHiveJava — CHANGES_REQUESTED** [review]

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeftSessionMessage.java (hunk 5 above)
Comment (Ranray) [reply Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeftSessionMessage.java:33]: Fixed in #7 .

**Review by BeeHiveJava — APPROVED** [review]

**Pull request merged**